### PR TITLE
Feature/Enhance ID3 Support: Implement APIC Functionality

### PR DESCRIPTION
**Pull request by fernandocQualabs**
## Description
- Added functionality for parsing APIC Id3 frames (id3 with images). 
- Added Shaka and HLS Id3 unit tests.

- Added new unit tests in CML:
-- readId3Size
-- isId3Header
-- isId3Footer
-- decodeId3Frame

### CODE REVIEW

File: lib/src/id3/util/decodeId3ImageFrame.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			const lowPart = baseCodePoint & 0x3ff;
+			decoded += String.fromCharCode(0xd800 + highPart);
+			decoded += String.fromCharCode(0xdc00 + lowPart);
+		}
+	}
+
+	return decoded;
+}
+
+type TypedArray = Int8Array | Uint8Array | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array | Uint8ClampedArray;
```
Comment (fpyoung) [inline lib/src/id3/util/decodeId3ImageFrame.ts]: Maybe we should add a comment here explaining why are we adding all these types on the first place

File: lib/test/structuredfield/decodeId3TextFrame.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { describe, it } from 'node:test';
+import { decodeId3TextFrame } from '../../src/id3/util/decodeId3TextFrame.js';
+import { Id3Frame } from '../../src/id3';
+
+describe('decodeId3TextFrame', () => {
+	it('should decode a TXXX frame', () => {
+		const frame = {
+			type: 'TXXX',
+			data: new Uint8Array([0, 102, 111, 111, 0, 97, 98, 99]),
+			size: 2, // required by the _decodeTextFrame function
```
Comment (fpyoung) [inline lib/test/structuredfield/decodeId3TextFrame.test.ts]: We should either:
- Remove the '_' in the _decodeTextFrame
- Remove the comment.

File: lib/src/id3/util/decodeId3ImageFrame.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	};
+
+	/**
+	 * Format:
+	 * [0]       = {Text Encoding}
+	 * [1 - X]   = {MIME Type}\0
+	 * [X+1]     = {Picture Type}
+	 * [X+2 - Y] = {Description}\0
+	 * [Y - ?]   = {Picture Data or Picture URL}
+	 */
```
Comment (fpyoung) [inline lib/src/id3/util/decodeId3ImageFrame.ts]: Do we want to keep this comment?

File: lib/src/id3/util/isId3Header.ts [hunk 4]
```diff
@@ -45,6 +45,5 @@ export function isId3Header(data: Uint8Array, offset: number): boolean {
 			}
 		}
 	}
-
```
Comment (fpyoung) [inline lib/src/id3/util/isId3Header.ts:48]: is this space a consecuence of linter?

**Review by hernanr99 — APPROVED** [review]
LGTM

**Review by fpyoung — APPROVED** [review]

File: lib/src/id3/util/decodeId3ImageFrame.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+	let uint8 = toUint8(data);
+	// If present, strip off the UTF-8 BOM.
+	if (uint8[0] == 0xef && uint8[1] == 0xbb && uint8[2] == 0xbf) {
+		uint8 = uint8.subarray(3);
+	}
+
+	// Homebrewed UTF-8 decoder based on
+	// https://en.wikipedia.org/wiki/UTF-8#Encoding
+	// Unlike decodeURIComponent, won't throw on bad encoding.
```
Comment (fpyoung) [inline lib/src/id3/util/decodeId3ImageFrame.ts]: Maybe we should remove this line if this function doesn't exist.

**Review by fpyoung — CHANGES_REQUESTED** [review]

**Review by fpyoung — APPROVED** [review]

**Review by mpollingerQualabs — APPROVED** [review]

File: scripts/createReleaseNotes.ts
Comment (mpollingerQualabs) [inline scripts/createReleaseNotes.ts:1]: Do we need to modify this file?

**Review by coatesjuan — APPROVED** [review]

**Comment by mpollingerQualabs** [comment]
Can this be closed/merged?

**Comment by fernandocQualabs** [comment]
This is already merged into the main @svta/common-media-library repo

**Pull request closed without merging**